src/main/java/com/google/cloud/tools/managedcloudsdk/internal/extract/ZipExtractor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.internal.extract;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.commons.compress.archivers.zip.ZipArchiveEntry;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+
+/**
+ * Extractor for *.zip google cloud sdk archives.
+ *
+ * <p>NOTE: this does not handle links or symlinks or any other kind of special types in the tar. It
+ * will only create files and directories. It also will NOT preserve file permissions.
+ */
+public class ZipExtractor implements Extractor {
+
+  private final Path archive;
+  private final Path destination;
+  private final ExtractorMessageListener listener;
+
+  ZipExtractor(Path archive, Path destination, ExtractorMessageListener listener) {
+    this.archive = archive;
+    this.destination = destination;
+    this.listener = listener;
+  }
+
+  @Override
+  public Path call() throws IOException {
```

Comment:
It seems that there is a lot of boilerplate code that is the same as in TarGzExtractor. Perhaps have an abstract skeletal implementation for both to inherit from? But probably unnecessary unless more archive types would be added.

Reply:
seemed fine before, but I think I'm going to restructure this if we're adding more checks like "check if actually extracted"